test-jest/chart/FunnelChart.spec.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { FunnelChart, Funnel } from '../../src';
+
+const data = [
+  { value: 100, name: '展现' },
+  { value: 80, name: '点击' },
+  { value: 50, name: '访问' },
+  { value: 40, name: '咨询' },
+  { value: 26, name: '订单' },
+];
+
+const data01 = [
+  { value: 100, name: '展现' },
+  { value: 80, name: '点击' },
+  { value: 50, name: '访问' },
+  { value: 40, name: '咨询' },
+  { value: 26, name: '订单' },
+];
+
+const data02 = [
+  { value: 60, name: '展现' },
+  { value: 50, name: '点击' },
+  { value: 30, name: '访问' },
+  { value: 20, name: '咨询' },
+  { value: 6, name: '订单' },
+];
+
+describe('<FunnelChart />', () => {
+  jest.useFakeTimers();
+
+  test('Renders 1 funnel in simple FunnelChart', () => {
+    const { container } = render(
+      <FunnelChart width={500} height={300}>
+        <Funnel dataKey="value" data={data} isAnimationActive={false} />
+      </FunnelChart>,
+    );
+    expect(container.querySelectorAll('.recharts-trapezoids')).toHaveLength(1);
+    expect(screen.getAllByRole('img')).toHaveLength(5);
+
+    data.forEach(({ name }) => {
+      expect(container.querySelector(`[name="${name}"]`)).toBeTruthy();
+    });
+  });
+
+  test('Renders 1 funnel in FunnelChart with animation', () => {
+    const { container } = render(
+      <FunnelChart width={500} height={300}>
+        <Funnel dataKey="value" data={data} isAnimationActive />
+      </FunnelChart>,
+    );
+
+    expect(container.querySelectorAll('.recharts-funnel-trapezoid')).toHaveLength(5);
+    // trapezoid are not rendered because the animation hasn't started
+    expect(container.querySelectorAll('.recharts-trapezoid')).toHaveLength(0);
+
+    // wait animation end
+    jest.advanceTimersByTime(500);
+
+    expect(container.querySelectorAll('.recharts-funnel-trapezoid')).toHaveLength(5);
+    // all trapezoids are visible
+    expect(container.querySelectorAll('.recharts-trapezoid')).toHaveLength(5);
+  });
+
+  test('Renders 2 funnel in nest FunnelChart', () => {
+    const { container } = render(
+      <FunnelChart margin={{ top: 20, right: 50, left: 20, bottom: 0 }} width={500} height={300}>
+        <Funnel dataKey="value" data={data01} isAnimationActive={false} />
+        <Funnel dataKey="value" data={data02} isAnimationActive={false} width="80%" />
+      </FunnelChart>,
+    );
+
+    expect(container.querySelectorAll('.recharts-trapezoids')).toHaveLength(2);
+    expect(screen.getAllByRole('img')).toHaveLength(10);
```

Comment:
What does the role img refer to?

Reply:
Each `.recharts-funnel-trapezoid` has a role img. 
Usually there I haven't find any role on the sag element so I think to use it.
Since it's not clear I changed it to: `container.querySelectorAll('.recharts-funnel-trapezoid')`

Reply:
yeah, this is interesting https://github.com/recharts/recharts/pull/2960

This added a bunch of role attrs in an attempt to address accessibility issues. We may need to revisit that in the future, why it was done, and how we can make it better